*: adjust dm docs structure and refine some expressions (#653)
pingcap/docs-dm: en/quick-start-with-dm.md

# Quick Start Guide for TiDB Data Migration

This document describes how to migrate data from MySQL to TiDB using [TiDB Data Migration](https://github.com/pingcap/dm) (DM).

## Sample scenario

> Suppose you deploy DM-master and DM-worker instances locally, and migrates data from an upstream MySQL instance to a downstream TiDB instance.

Reviewer: ```suggestion
Suppose you deploy DM-master and DM-worker instances locally, and migrate data from an upstream MySQL instance to a downstream TiDB instance.
```

{{< copyable "shell-regular" >}}

```bash
./bin/dmctl --encrypt "123456"
```

```
fCxfQ9XKCezSzuCD0Wf5dUD+LsKegSg=
```

Save this encrypted value, and use it for creating a MySQL data source in the following steps.

#### Edit the source configuration file of the MySQL instance

Write the following configurations to `conf/source1.yaml`.

```yaml
# MySQL1 Configuration.
source-id: "mysql-replica-01"
from:
  host: "127.0.0.1"
  user: "root"
  password: "fCxfQ9XKCezSzuCD0Wf5dUD+LsKegSg="
  port: 3306
```

#### Load data source configuration file

To load the data source configuration file of MySQL to the DM cluster using dmctl, run the following command in the terminal:

{{< copyable "shell-regular" >}}

```bash
./bin/dmctl --master-addr=127.0.0.1:8261 operate-source create mysql-source-conf.yaml
```

The following is an example of the returned results:

```bash
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "worker1"
        }
    ]
}
```

Now you successfully add the data source `MySQL-3306` to the DM cluster.

### Create a data migration task

> After inserting the [sample data](#prepare-sample-data) into `MySQL-3306`, take the following two steps to migrate the table `testdm`.`t1` and `testdm`.`t2` to the downstream TiDB instance:

Reviewer: Suggest removing "two" from the description, because if three or more steps are needed for this task in the future, we might forget to update "two".

```suggestion
After inserting the [sample data](#prepare-sample-data) into `MySQL-3306`, take the following steps to migrate the tables `testdm`.`t1` and `testdm`.`t2` to the downstream TiDB instance:
```

> 1. Write the following configurations to `testdm-task.yaml` to create a task configuration file:

Reviewer: ```suggestion
1. Create a task configuration file `testdm-task.yaml`, and add the following configurations to the file.
```

    {{< copyable "" >}}

    ```yaml
    ---
    name: testdm
    task-mode: all
    target-database:
    host: "127.0.0.1"
    port: 4000
    user: "root"
    password: "" # If the password is not null, it is recommended to use password encrypted with dmctl.
    mysql-instances:

>     - source-id: "mysql-replica-01"

Reviewer 2: ```suggestion
      - source-id: "mysql-replica-01"
```

        block-allow-list:  "ba-rule1"
    block-allow-list:
    ba-rule1:
        do-dbs: ["testdm"]
    ```

2. Load the task configuration file using dmctl:

    {{< copyable "shell-regular" >}}

    ```bash
    ./bin/dmctl -master-addr 127.0.0.1:8261 start-task testdm-task.yaml
    ```

    The following is an example of the returned results:

    ```bash
    {
        "result": true,
        "msg": "",
        "sources": [
            {
                "result": true,
                "msg": "",
                "source": "mysql-replica-01",
                "worker": "worker1"
            }
        ]
    }
    ```

> Now you successfully create a data migration task that migrates date from `MySQL-3306` to the downstream TiDB instance.

Reviewer: ```suggestion
Now you successfully create a data migration task that migrates data from `MySQL-3306` to the downstream TiDB instance.
```

> ### Check data migration task status

Reviewer: ```suggestion
### Check status of the data migration task 
```
